wcfsetup/install/files/lib/system/cache/source/DiskCacheSource.class.php

```diff
@@ -157,4 +157,20 @@ private function readCache(string $filename): mixed
             throw new \Exception("Failed to unserialize the cache contents.", 0, $e);
         }
     }
+
+    #[\Override]
+    public function getCacheLifetime(string $cacheName): ?int
+    {
+        $filename = $this->getFilename($cacheName);
+
+        if (!\file_exists($filename)) {
+            return null;
+        }
+
+        if (!@\filesize($filename)) {
+            return null;
+        }
+
+        return \filemtime($filename);
```

Comment:
How does the `\filemtime()` value represent the maximum lifetime of a file? You’re returning a timestamp here but in Redis the TTL in seconds is provided instead.